### PR TITLE
added pip requirment for config virtualenv.create=False

### PR DESCRIPTION
**Pull request by KGB33**
Resolves: #1165

- [x] Added **tests** for changed code. -- NA - no code changed
- [x] Updated **documentation** for changed code. -- NA - no code changed

### CODE REVIEW

**Comment by abn** [comment]
This should no longer be required since #2666 right? The embedded pip from `virtualenv` is used.

**Comment by KGB33** [comment]
For installing project dependencies that makes sense but how does poetry install it's `_vendor` packages? 

The `_vendor` folder in the (latest) release on Github is empty.

**Comment by abn** [comment]
From `poetry-1.1.0b2-linux.tar.gz` the `poetry/_vendor` direcotry is already populated with the vendored dependencies.

```console
$ ls poetry/_vendor
py2.7  py3.5  py3.6  py3.7  py3.8  py3.9
```

At the moment, the `get-poetry.py` script does not rely on `pip` or any external dependnecies other than a valid python version. Note that the vendor directory is only valid for when installed using the custom installer (`get-poetry.py`). For installation using `pip` or `pipx` dependencies are managed by the package management frontend.

**Comment by KGB33** [comment]
Gotcha. I checked the source code zip and just incorrectly assumed the `_vender` folder was un-populated  for all of them.

**Comment by abn** [comment]
@KGB33 this still has value; but maybe make it clear that this is only required for cases where the project is using the system environment (`poetry config virtualenvs.create false`). And maybe this is better off in the "Configuration" section of the docs.

**Comment by KGB33** [comment]
@abn Sorry for the delay, I've added a simple note to the configuration docs. I was thinking adding an error message might be helpful as well, however, I haven't gotten a docker image to work for me quite yet.

**Comment by KGB33** [comment]
Got the docker image to work for me and the error message is very clear. 

```
Skipping virtualenv creation, as specified in config file.
Installing dependencies from lock file

Package operations: 71 installs, 0 updates, 0 removals

  • Installing pycparser (2.20)

  EnvCommandError

  Command ['/usr/local/bin/python3', '-m', 'pip', 'install', '--no-deps', '/root/.cache/pypoetry/artifacts/37/8e/5a/0ea4f84bc7f11e0e3468110efa2c7783241ea7eaa63a92a751de06f78f/pycparser-2.20-py2.py3-none-any.whl'] errored with the following return code 1, and output:
  /usr/local/bin/python3: No module named pip


  at /poetry_install/libutils/env.py:915 in _run
       911│                 output = subprocess.check_output(
       912│                     cmd, stderr=subprocess.STDOUT, **kwargs
       913│                 )
       914│         except CalledProcessError as e:
    →  915│             raise EnvCommandError(e, input=input_)
       916│
       917│         return decode(output)
       918│
       919│     def execute(self, bin, *args, **kwargs):
```

File: docs/docs/configuration.md [hunk 1]
```diff
@@ -107,6 +107,10 @@ Defaults to one of the following directories:
 Create a new virtual environment if one doesn't already exist.
 Defaults to `true`.
 
+!!!note:
+	While `pip` is not a system requirment for poetry when installed via the provided installer,
+	`pip` is required if this is set to `false`.
```
Comment (abn) [inline docs/docs/configuration.md:112]:
```suggestion
        When setting this configuration to `false`, the Python environment used must have `pip` 
        installed and available.
```

Rephrasing it a bit to make it clear(er) - I think. :)

**Comment by KGB33** [comment]
What about adding why. Something like:

```
!!!note:
        When setting this configuration to `false`, the Python environment used must have `pip` 
        installed and available because Poetry cannot use the vendored `virtualenv`. 

````

**Review by abn — APPROVED** [review]

**Pull request merged**